src/cdn_definitions/_impl/__init__.py

```diff
@@ -1,26 +1,103 @@
 import json
 import os
+import warnings
+
+import requests
 import yaml
+from urllib3.util.retry import Retry
+
 
+def to_dict(data, ext):
+    """
+    Converts a string or a requests.Response object to a dict.
+    """
+    if ext.lower() == ".json":
+        return json.loads(data.json() if isinstance(data, requests.Response) else data)
```

Comment:
Doesn't `data.json()` already parse the content, so you're trying to parse it twice here?

If I'm right then you'd probably want to also look into how it is that the tests could pass while this bug exists.